### PR TITLE
soundwire hardening

### PR DESCRIPTION
**Pull request by plbossart**
This is the take2 on interrupt filtering, cleaner handing of clock stop and tentative fix for multpipeline stress test.

This PR replaces #1640 and should also replace #1654 

Issues to be checked: #1494 #1637 #1638 #1660

### CODE REVIEW

File: drivers/soundwire/intel.c [hunk 1]
```diff
@@ -1722,6 +1722,9 @@ static int intel_resume_runtime(struct device *dev)
 	clock_stop_quirks = sdw->link_res->clock_stop_quirks;
 
 	if (clock_stop_quirks & SDW_INTEL_CLK_STOP_TEARDOWN) {
+
```
Comment (lyakh) [inline drivers/soundwire/intel.c:1725]: nitpick: I wouldn't add empty lines at the beginning of `if () {}` and similar blocks - in this and following commits.

File: drivers/soundwire/stream.c [hunk 2]
```diff
@@ -635,12 +635,17 @@ static int sdw_notify_config(struct sdw_master_runtime *m_rt)
...
  */
-static int sdw_program_params(struct sdw_bus *bus)
+static int sdw_program_params(struct sdw_bus *bus, bool prepare)
 {
 	struct sdw_master_runtime *m_rt;
 	int ret = 0;
 
 	list_for_each_entry(m_rt, &bus->m_rt_list, bus_node) {
+
+		if (!prepare &&
```
Comment (lgirdwood) [inline drivers/soundwire/stream.c:651]: nitpick: this could do with a comment explaining the state check.

Comment (plbossart) [reply drivers/soundwire/stream.c:651]: yes, we need to explain this better, will fix.

File: drivers/soundwire/intel.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/intel.c:1725]: I leave this as a degree of freedom, others use empty lines for for loops as well, it's just easier to identify the body and compare different cases.

**Comment by plbossart** [comment]
Thanks @YvonneYang2 and @RanderWang, looks like we killed 3 nasty bugs with this PR. Will update and merge. Good progress!

**Comment by plbossart** [comment]
patches applied on integration/soundwire-intel and merged on integration/soundwire-latest.

That should take care of all the complicated bugs, until new ones are found of course

**Pull request closed without merging**